compiler/hash-parser/src/lexer.rs

```diff
@@ -0,0 +1,618 @@
+                // start as identifier).
+                c if is_id_start(c) => break self.ident(),
+
+                // Numeric literal.
+                '0'..='9' => {
+                    break self
+                        .number()
+                        .unwrap_or_else(|e| panic!("error: {:#?}", e.message))
+                }
+                // character literal.
+                '\'' => {
+                    break self
+                        .char()
+                        .unwrap_or_else(|e| panic!("error: {:#?}", e.message)); // @@ErrorReporting: this is where we hook into error reporting to print the result
+                                                                                // and display it to the user
+                }
+
+                // String literal.
+                '"' => break self.string().unwrap_or_else(|e| panic!("error: {:?}", e)),
+
+                _ => break TokenKind::Unexpected,
+            }
+        };
+
+        let location = Location::span(offset, self.len_consumed());
+        Some(Token::new(token_kind, location))
+    }
+
+    /// Consume an identifier, at this stage keywords are also considered to be identfiers
+    pub(crate) fn ident(&self) -> TokenKind {
+        debug_assert!(is_id_start(self.prev.get()));
+
+        let first = self.prev.get();
+        let suffix = self.eat_while(is_id_continue);
+        let name: String = iter::once(first).chain(suffix).collect();
+
+        // create the identifier here from the created map
+        let ident = IDENTIFIER_MAP.create_ident(AstString::Owned(name.as_str().to_owned()));
+        TokenKind::Ident(ident)
+    }
+
+    /// Consume a number literal, either float or integer
+    pub(crate) fn number(&self) -> TokenResult<TokenKind> {
+        let prev = self.prev.get();
+        debug_assert!(('0'..='9').contains(&prev));
+
+        // record the start location of the literal
+        let start = self.offset.get() - 1;
+
+        // firstly, figure out if this literal has a base, if so then we need to perform
+        // some magic at the end to cast it into the correct base...
+        if prev == '0' {
+            let maybe_radix = match self.peek() {
+                'b' => Some(2),
+                'o' => Some(8),
+                'x' => Some(16),
+                _ => None,
+            };
+
+            // if this does have a radix then we need to handle the radix
+            if let Some(radix) = maybe_radix {
+                self.next(); // accounting for the radix
+
+                let chars = self.eat_while(|c| c.is_digit(radix)).collect::<String>();
+                let value = u64::from_str_radix(chars.as_str(), radix);
+
+                // @@ErrorHandling: We shouldn't error here, this should be handeled by the SmallVec<..> change to integers
+                if value.is_err() {
+                    println!("value={}, radix={}", chars, radix);
```

Comment:
Should be turned to debug?

Reply:
This is a bogus `println!`, should be removed.